packages/core/src/checkbox/Checkbox.tsx

```diff
@@ -77,7 +78,19 @@ export interface CheckboxProps
   /**
    * Validation status.
    */
-  validationStatus?: "error" | "warning";
+  validationStatus?: CheckboxValidationStatus;
+}
+
+function getValidationStatus(
+  checkboxGroupStatus: CheckboxValidationStatus,
+  formFieldStatus: CheckboxValidationStatus | "success" | undefined,
+  checkboxStatus: CheckboxValidationStatus
+) {
+  return checkboxGroupStatus ?? formFieldStatus
+    ? formFieldStatus !== "success"
+      ? formFieldStatus
+      : undefined
+    : checkboxStatus;
```

Comment:
Not sure I understand the logic here. Can you please add a comment?
Why do we allow for 'success' status if we're ignoring it?